bin/lib/library_builder.py

```diff
@@ -586,26 +598,18 @@ def is_already_uploaded(self, buildfolder):
 
     def set_as_uploaded(self, buildfolder):
         conanhash = self.get_conan_hash(buildfolder)
-        if conanhash == None:
+        if conanhash is None:
             raise RuntimeError(f'Error determining conan hash in {buildfolder}')
 
         self.logger.info(f'commithash: {conanhash}')
 
         annotations = self.get_build_annotations(buildfolder)
-        if not 'commithash' in annotations:
+        if 'commithash' not in annotations:
             self.upload_builds()
         annotations['commithash'] = self.get_commit_hash()
 
-        for lib in self.buildconfig.staticliblink:
-            if os.path.exists(os.path.join(buildfolder, f'lib{lib}.a')):
-                bininfo = BinaryInfo(self.logger, buildfolder, os.path.join(buildfolder, f'lib{lib}.a'))
-                libinfo = bininfo.cxx_info_from_binary()
-                archinfo = bininfo.arch_info_from_binary()
-                annotations['cxx11'] = libinfo['has_maybecxx11abi']
-                annotations['machine'] = archinfo['elf_machine']
-                annotations['osabi'] = archinfo['elf_osabi']
-
-        for lib in self.buildconfig.sharedliblink:
+        for lib in itertools.chain(self.buildconfig.staticliblink, self.buildconfig.sharedliblink):
+            # TODO - this is the same as the original code but I wonder if this needs to be *.so for shared?
```

Comment:
Oh, hmm, maybe yes.. haven't really tested shared libraries with this, so could very well be